src/main/scala/io/citrine/lolo/transformers/Standardizer.scala

```diff
@@ -0,0 +1,221 @@
+}
+
+/**
+  * Model that wrapps the base model next to the transformations
+  *
+  * @param baseModel
+  * @param trans
+  * @tparam T
+  */
+class StandardizerModel[T](baseModel: Model[PredictionResult[T]], trans: Seq[Option[(Double, Double)]]) extends Model[PredictionResult[T]] {
+
+  /**
+    * Standardize the inputs and then apply the base model
+    *
+    * @param inputs to apply the model to
+    * @return a predictionresult which includes, at least, the expected outputs
+    */
+  override def transform(inputs: Seq[Vector[Any]]): StandardizerPrediction[T] = {
+    val standardInputs = Standardizer.applyStandardization(inputs, trans.tail)
+    new StandardizerPrediction(baseModel.transform(standardInputs), trans)
+  }
+}
+
+/**
+  * Prediction that wraps the base prediction next to the transformation
+  *
+  * @param baseResult
+  * @param trans
+  * @tparam T
+  */
+class StandardizerPrediction[T](baseResult: PredictionResult[T], trans: Seq[Option[(Double, Double)]]) extends PredictionResult[T] {
+  /**
+    * Get the expected values for this prediction
+    *
+    * Just reverse any transformation that was applied to the labels
+    *
+    * @return expected value of each prediction
+    */
+  override def getExpected(): Seq[T] = {
+    baseResult.getExpected().map {
+      case x: Double => x * rescale + intercept
+      case x: Any => x
+    }.asInstanceOf[Seq[T]]
+  }
+
+  /**
+    * Get the uncertainty of the prediction
+    *
+    * This is un-standardized by rescaling by the label's variance
+    *
+    * @return uncertainty of each prediction
+    */
+  override def getUncertainty(): Option[Seq[Any]] = {
+    baseResult.getUncertainty() match {
+      case Some(x) if trans.head.isDefined => Some(x.map(_.asInstanceOf[Double] * rescale))
+      case x: Any => x
+    }
+  }
+
+
+  /**
+    * Get the gradient or sensitivity of each prediction
+    *
+    * This is un-stanardized by rescaling by the label's variance divided by the feature transformation's variance
+    *
+    * @return a vector of doubles for each prediction
+    */
+  override def getGradient(): Option[Seq[Vector[Double]]] = {
+    baseResult.getGradient() match {
+      case None => None
+      case Some(x) =>
+        Some(x.map(g => g.zip(trans.tail).map {
+          // If there was a (linear) transformer used on that input, take the slope "m" and rescale by it
+          case (y: Double, Some((_, m))) => y * rescale * m
+          // Otherwise, just rescale by the output transformer
+          case (y, None) => y * rescale
+        }))
+    }
+  }
+
+  val rescale = 1.0 / trans.head.map(_._2).getOrElse(1.0)
+  val intercept = trans.head.map(_._1).getOrElse(1.0)
+}
+
+/**
+  * Utilities to compute and apply standarizations
+  */
+object Standardizer {
+
+  /**
+    * The standardizations are just shifts and rescale.  The shift is by the mean and the re-scale is by the variance
```

Comment:
It might be nice to think about extensibility for different types of scalers (e.g. min-max scalers.)